kore.maude

```diff
@@ -220,18 +227,30 @@ Questions about this definition:
 - do we really need ~> as a K construct?  Can it be a label and thus the way K
   deal with evaluation contexts just a particular methodology and not an
   intrisic part of K ? Same question for =>.
+  Cosmin: Syntacticly at least, I would say "no" for both. From the implementation
+  perspective, this will simplify traversal code even further. 
 - rename .::K and .::KList to . ?  if we get rid of ~> as above, then we may
   only need .::KList.
+  Cosmin: I agree.
 - Should we allow empty productions instead of sort declarations?  A pro
   argument is that we have fewer syntactic constructs.  A cons argument is
   that a sort declaration may be regarded as a different entityr, with
   different kinds of attributes.
+  Cosmin: I don't quite follow.
 - Should we add constant AttributeKeys which have special semantics, like
   "strict", as constants?
+  Cosmin: we may want to discuss the attribute syntax separately.
 - Brackets and disappearing injections are the same thing, arent's they?  A
   production which only has one non-terminal and the production disappears
   from the AST.
+  Cosmin: Yes, I also think they can be seen as the same thing. That being
+          said, I would really like for injections to not dissapear. If something
+          is meant to dissapear, it should be declared as a subsort. And the
+          dissapearing injections (including subsorts) could be done via rules as
+          we discussed before.
```

Comment:
I've stared arguing for this again in the previous message. I agree it would be nice to do without sorts but the reality is that we do have them now. Keeping them as second-hand citizens for which magic things happen is confusing. Changing the default behavior of `syntax A ::= B` from subsorting to a regular production would help. 

> Can we for now use it as an attribute to productions that can be subsorts?

I'm a bit worried about adding semantics to the attributes, but it could work as an intermediate solution from my perspective.